cdap-docs/reference-manual/source/release-notes.rst

```diff
@@ -23,6 +23,307 @@ Cask Data Application Platform Release Notes
+  UI widgets for the Hydrator Kafka real-time source, JMS real-time source, and CloneRecord
+  transform.
+
+- `CDAP-4858 <https://issues.cask.co/browse/CDAP-4858>`__ - Fixed a problem with the CDAP
+  UI not correctly cleaning up upon namespace deletion.
+
+- `CDAP-4868 <https://issues.cask.co/browse/CDAP-4868>`__ - Fixed a problem with the run
+  record history and the last runtime of pipeline not matching.
+
+- `CDAP-4870 <https://issues.cask.co/browse/CDAP-4870>`__ - Fixed a problem with the
+  Teradata plugin not displaying the column case field with a drop-down menu widget.
+
+- `CDAP-4873 <https://issues.cask.co/browse/CDAP-4873>`__ - Fixed a problem with starting
+  the CDAP Standalone SDK under Macintosh OS X in a directory whose path has a space.
+
+- `CDAP-4877 <https://issues.cask.co/browse/CDAP-4877>`__ - Fixed errors in Cask Hydrator
+  Plugins. Two plugin documents (core-plugins/docs/Database-batchsink.md and
+  core-plugins/docs/Database-batchsource.md) were removed, as the plugins have been moved
+  from core-plugins to database-plugins (to database-plugins/docs/Database-batchsink.md and
+  database-plugins/docs/Database-batchsource.md).
+
+- `CDAP-4882 <https://issues.cask.co/browse/CDAP-4882>`__ - Fixed a problem with searching
+  for plugins in the Hydrator plugin causing formatting issues when multiple tabs are open.
+
+- `CDAP-4886 <https://issues.cask.co/browse/CDAP-4886>`__ - Fixed a problem where the
+  suppression of uncaught exceptions resulted in the exception not being logged at all. These
+  exceptions will now be logged, even at the risk of being logged more than once.
+
+- `CDAP-4887 <https://issues.cask.co/browse/CDAP-4887>`__ - Fixed a problem with the CDAP
+  Upgrade Tool raising exceptions when it encountered a non-CDAP table in HBase.
+
+- `CDAP-4889 <https://issues.cask.co/browse/CDAP-4889>`__ - Fixed an issue with upgrading
+  HBase tables while using the CDAP Upgrade Tool.
+
+- `CDAP-4894 <https://issues.cask.co/browse/CDAP-4894>`__ - Fixed an issue with CDAP
+  coprocessors that caused HBase tables to be disabled after upgrading the cluster to a
+  highly-available file system.
+
+- `CDAP-4905 <https://issues.cask.co/browse/CDAP-4905>`__ - Fixed a problem in the
+  documentation with the example of using Cask Hydrator with third-party JARs.
+
+- `CDAP-4906 <https://issues.cask.co/browse/CDAP-4906>`__ - Fixed the CDAP Upgrade Tool to
+  return a non-zero exit status upon error during upgrade.
+
+Deprecated and Removed Features
+-------------------------------
+
+- See the :ref:`CDAP 3.3.1 Javadocs <javadocs>` for a list of deprecated and removed APIs.
+
+- The properties ``router.ssl.webapp.bind.port``, ``router.webapp.bind.port``,
+  ``router.webapp.enabled`` have been deprecated and will be removed in a future version.
+
+.. _known-issues-331:
+
+Known Issues
+------------
+- `CDAP-4789 <https://issues.cask.co/browse/CDAP-4789>`__ - There is an issue with the
```

Comment:
Why don't we leave this PR open until that is done and merged. We can update the release note then...